Exporting note as PDF works but resolving URI crashes

### What happened?

I exported a note as PDF to the standard download folder. The note conatined only a QR code as PNG and the link in text form. The PDF was successfully created and looks okay but NotallyX crashed right after.



### App Version

7.6.1

### Android Version (API Level)

35

### (Optional) Relevant log output

```shell
org.chromium.base.JniAndroid$UncaughtExceptionException: Native stack trace:
#00 pc 0x0000000006872f6b /data/app/~~q_BBL1T0ACTyTrdfj5P9Tw==/com.google.android.trichromelibrary_763200743-XuiHe_Ym0rdbOZA4c2PmdQ==/base.apk (offset 0x930000)
#01 pc 0x000000000585100b /data/app/~~q_BBL1T0ACTyTrdfj5P9Tw==/com.google.android.trichromelibrary_763200743-XuiHe_Ym0rdbOZA4c2PmdQ==/base.apk (offset 0x930000)
#02 pc 0x000000000227e4ef /data/app/~~q_BBL1T0ACTyTrdfj5P9Tw==/com.google.android.trichromelibrary_763200743-XuiHe_Ym0rdbOZA4c2PmdQ==/base.apk (offset 0x930000)
#03 pc 0x0000000006ea611f /data/app/~~q_BBL1T0ACTyTrdfj5P9Tw==/com.google.android.trichromelibrary_763200743-XuiHe_Ym0rdbOZA4c2PmdQ==/base.apk (offset 0x930000)
#04 pc 0x0000000002282c9b /data/app/~~q_BBL1T0ACTyTrdfj5P9Tw==/com.google.android.trichromelibrary_763200743-XuiHe_Ym0rdbOZA4c2PmdQ==/base.apk (offset 0x930000)
#05 pc 0x0000000006cf5613 /data/app/~~q_BBL1T0ACTyTrdfj5P9Tw==/com.google.android.trichromelibrary_763200743-XuiHe_Ym0rdbOZA4c2PmdQ==/base.apk (offset 0x930000)
#06 pc 0x0000000006eb1387 /data/app/~~q_BBL1T0ACTyTrdfj5P9Tw==/com.google.android.trichromelibrary_763200743-XuiHe_Ym0rdbOZA4c2PmdQ==/base.apk (offset 0x930000)
#07 pc 0x000000000690be57 /data/app/~~q_BBL1T0ACTyTrdfj5P9Tw==/com.google.android.trichromelibrary_763200743-XuiHe_Ym0rdbOZA4c2PmdQ==/base.apk (offset 0x930000)
#08 pc 0x000000000752f69f /data/app/~~q_BBL1T0ACTyTrdfj5P9Tw==/com.google.android.trichromelibrary_763200743-XuiHe_Ym0rdbOZA4c2PmdQ==/base.apk (offset 0x930000)
#09 pc 0x0000000006a91193 /data/app/~~q_BBL1T0ACTyTrdfj5P9Tw==/com.google.android.trichromelibrary_763200743-XuiHe_Ym0rdbOZA4c2PmdQ==/base.apk (offset 0x930000)
#10 pc 0x0000000006a9110f /data/app/~~q_BBL1T0ACTyTrdfj5P9Tw==/com.google.android.trichromelibrary_763200743-XuiHe_Ym0rdbOZA4c2PmdQ==/base.apk (offset 0x930000)
#11 pc 0x00000000076114b7 /data/app/~~q_BBL1T0ACTyTrdfj5P9Tw==/com.google.android.trichromelibrary_763200743-XuiHe_Ym0rdbOZA4c2PmdQ==/base.apk (offset 0x930000)
#12 pc 0x00000000000087f3 /system/lib64/libutils.so
#13 pc 0x00000000000def33 /system/lib64/libandroid_runtime.so

	at org.chromium.base.JniAndroid.handleException(chromium-TrichromeWebViewGoogle6432.aab-beta-763200743:21)
	at android.os.MessageQueue.nativePollOnce(Native Method)
	at android.os.MessageQueue.next(MessageQueue.java:387)
	at android.os.Looper.loopOnce(Looper.java:189)
	at android.os.Looper.loop(Looper.java:317)
	at android.app.ActivityThread.main(ActivityThread.java:8935)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:594)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:911)
Caused by: java.lang.IllegalArgumentException: Invalid URI: content://com.android.providers.downloads.documents/document/15
	at android.provider.DocumentsContract.getTreeDocumentId(DocumentsContract.java:1289)
	at androidx.documentfile.provider.DocumentFile.fromTreeUri(SourceFile:3)
	at com.philkes.notallyx.utils.AndroidExtensionsKt.toReadablePath(SourceFile:30)
	at com.google.crypto.tink.KeysetHandle.onSuccess(SourceFile:33)
	at androidx.appcompat.widget.AppCompatDrawableManager$1.onSuccess(SourceFile:117)
	at android.print.PdfExtensionsKt$writeToFile$onWriteResult$1.onWriteFinished(SourceFile:5)
	at WV.vk.a(chromium-TrichromeWebViewGoogle6432.aab-beta-763200743:32)
	at org.chromium.android_webview.AwPdfExporter.didExportPdf(chromium-TrichromeWebViewGoogle6432.aab-beta-763200743:3)
	... 8 more
```

--- a/app/src/main/java/com/philkes/notallyx/utils/AndroidExtensions.kt
+++ b/app/src/main/java/com/philkes/notallyx/utils/AndroidExtensions.kt
@@ -533,7 +533,13 @@ fun Context.toReadablePath(uri: Uri): String {
     if (uri.authority == "com.android.externalstorage.documents") {
         return toReadable(uri.path!!)
     }
-    val documentFile = DocumentFile.fromTreeUri(this, uri)
+
+    val documentFile =
+        try {
+            DocumentFile.fromTreeUri(this, uri)
+        } catch (_: Exception) {
+            null
+        }
     return documentFile?.name?.let { "${uri.authority}/$it" }
         ?: uri.path?.let { toReadable(it) }
         ?: uri.toString()

